credentialproviderpackage/pkg/configurator/bottlerocket/bottlerocket_test.go

```diff
@@ -0,0 +1,339 @@
+package bottlerocket
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"credential-provider/pkg/constants"
+)
+
+type response struct {
+	statusCode   int
+	expectedBody []byte
+	responseMsg  string
+}
+
+func Test_bottleRocket_CommitChanges(t *testing.T) {
+	type fields struct {
+		client  http.Client
+		baseURL string
+		config  constants.CredentialProviderConfigOptions
+	}
+
+	tests := []struct {
+		name     string
+		fields   fields
+		wantErr  bool
+		response response
+		expected string
+	}{
+		{
+			name: "test success",
+			fields: fields{
+				client: http.Client{},
+				config: constants.CredentialProviderConfigOptions{
+					ImagePatterns:        []string{constants.DefaultImagePattern},
+					DefaultCacheDuration: constants.DefaultCacheDuration,
+				},
+			},
+			wantErr: false,
+			response: response{
+				statusCode:  http.StatusOK,
+				responseMsg: "",
+			},
+		},
+		{
+			name: "test fail",
+			fields: fields{
+				client: http.Client{},
+				config: constants.CredentialProviderConfigOptions{
+					ImagePatterns:        []string{constants.DefaultImagePattern},
+					DefaultCacheDuration: constants.DefaultCacheDuration,
+				},
+			},
+			wantErr: true,
+			response: response{
+				statusCode:  http.StatusNotFound,
+				responseMsg: "",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(tt.response.statusCode)
+				fmt.Fprintf(w, tt.response.responseMsg)
+			}))
+			b := &bottleRocket{
+				client:  tt.fields.client,
+				baseURL: svr.URL + "/",
+				config:  tt.fields.config,
+			}
+			if err := b.CommitChanges(); (err != nil) != tt.wantErr {
+				t.Errorf("UpdateAWSCredentials() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func Test_bottleRocket_UpdateAWSCredentials(t *testing.T) {
+	type fields struct {
+		client  http.Client
+		baseURL string
+		config  constants.CredentialProviderConfigOptions
+	}
+	type args struct {
+		path    string
+		profile string
+	}
+
+	tests := []struct {
+		name           string
+		fields         fields
+		args           args
+		patchResponse  response
+		commitResponse response
+		wantErr        bool
+	}{
+		{
+			name: "working credential update",
+			fields: fields{
+				client: http.Client{},
+				config: constants.CredentialProviderConfigOptions{},
+			},
+			args: args{
+				path:    "testdata/testcreds",
+				profile: "eksa-packages",
+			},
+			patchResponse: response{
+				statusCode:   http.StatusNoContent,
+				expectedBody: []byte("{\"aws\":{\"config\":\"W3Byb2ZpbGUgZWtzYS1wYWNrYWdlc10KYXdzX2FjY2Vzc19rZXlfaWQ9QUtJQUlPU0ZPRE5ON0VYQU1QTEUKYXdzX3NlY3JldF9hY2Nlc3Nfa2V5PXdKYWxyWFV0bkZFTUkvSzdNREVORy9iUHhSZmlDWUVYQU1QTEVLRVk=\",\"profile\":\"eksa-packages\",\"region\":\"\"}}"),
```

Comment:
Consider decoding this and calling the encode function before creating this object.